### DISCUSSION TITLE
Custom Reports: Pie Chart Look

### DISCUSSION THREAD

**Discussion by valsamis-d** [moneymanagerex/moneymanagerex#3178] [Q&A] [answered]
Hello,

First off, I hope I'm posting to the right place. If not please let me know where I could direct my question.

Next, I wanted to thank you guys thank for this amazing application! I love! I noticed there is somewhere a 'sponsors' page and would like to make a donation but I'm not sure how. Please, let me know.

Now, in regards to my request:

I have created some 'average' reports and would like to change the Pie Chart look to the one used in the OoB reports in v1.5.x but I'm not sure how to do that. If I remember correctly it was @vomikan who had helped me a few years ago with setting up "Lua" and "Template" parts, and now I'm not sure what exactly I have to change in order to use the new pie charts. I have also attached the '.grm' file.

[Average Spending per Category & Subcategory (To Date) - SQL Based.zip](https://github.com/moneymanagerex/moneymanagerex/files/6235768/Average.Spending.per.Category.Subcategory.To.Date.-.SQL.Based.zip)

I'd appreciate any pointers/help.

Thank you
Dimitrios

**Comment by vomikan** [comment, marked as answer]
Hello,

>I wanted to thank you guys thank for this amazing application! 

TRhank you for your feedback. 

![image](https://user-images.githubusercontent.com/6836805/113137803-abe49500-922d-11eb-8786-17089d0070fa.png)

https://www.moneymanagerex.org/development/donate

**Reply by valsamis-d** [reply]
MoneyManageEx team, enjoy the coffee :)

**Comment by vomikan** [comment]
I haven't done any reports with the new library yet. But I think it's even easier than it was before.

**Reply by valsamis-d** [reply]
Any idea who I could reach out for some help/assistance? If I could have access to "Lua" and "Template" parts of one of the default reports (EDIT: preferably the Categories > Where the Money Goes" one), I believe I could manage to adjust mine.

**Reply by vomikan** [reply]
[Average Spending per Category & Subcategory.grm.zip](https://github.com/moneymanagerex/moneymanagerex/files/6238261/Average.Spending.per.Category.Subcategory.grm.zip)

**Reply by valsamis-d** [reply]
> 
> 
> [Average Spending per Category & Subcategory.grm.zip](https://github.com/moneymanagerex/moneymanagerex/files/6238261/Average.Spending.per.Category.Subcategory.grm.zip)

Thank you so much @vomikan !

**Reply by valsamis-d** [reply]
@vomikan tested it with RC2. All was great. Just corrected the 'template' a bit do that columns "Total Spend" and "Average Spending" are formatted as currencies:

[TEMPLATE.zip](https://github.com/moneymanagerex/moneymanagerex/files/6244243/TEMPLATE.zip)

Also, I noticed that when hovering over the pie chart, labels are not displayed correctly:

![image](https://user-images.githubusercontent.com/17590539/113309145-8465f900-930f-11eb-9a75-b78c5f84383d.png)

I thought this was addressed sometime ago?

**Reply by vomikan** [reply]
https://github.com/moneymanagerex/moneymanagerex/issues/3075

**Reply by valsamis-d** [reply]
> https://github.com/moneymanagerex/moneymanagerex/issues/3075

It is shown as closed though. Tested it also with the lastest 1.5.1 Alpha 1 and still I'm getting the same behavior.

**Reply by vomikan** [reply]
Please update the html template
```
<script>

var chart_apex18467 = [
<TMPL_LOOP NAME=CONTENTS><TMPL_VAR "Average Spending">,</TMPL_LOOP>
 ]; 
var options = { chart: { type: 'pie', toolbar: { tools: { download: false } }, width: '100%' }, title: { text: ''}, 
  plotOptions: { pie: { customScale: 0.8 } },
  tooltip: { theme: 'dark' , y: { formatter: function(value, opts) { return chart_apex18467[opts.dataPointIndex] }
 } }
, dataLabels: { enabled: true, style: { fontSize: '16px' }, dropShadow: { enabled: false } }
, colors: [
'#008FFB','#00E396','#FEB019','#FF4560','#775DD0','#3F51B5','#03A9F4','#4CAF50','#F9CE1D','#FF9800','#33B2DF','#546E7A','#D4526E','#13D8AA','#A5978B','#4ECDC4','#81D4FA','#546E7A','#FD6A6A','#2B908F','#F9A3A4','#90EE7E','#FA4443','#69D2E7','#449DD1','#F86624'

],labels: [

<TMPL_LOOP NAME=CONTENTS>'<TMPL_VAR "Category Name">',</TMPL_LOOP>

], series: [

<TMPL_LOOP NAME=CONTENTS><TMPL_VAR "Average Spending">,</TMPL_LOOP>

] };
var chart = new ApexCharts(document.querySelector('#apex18467'), options); chart.render();

</script>
```

**Reply by valsamis-d** [reply]
Perfect :) Thank you!!
